### PR TITLE
Dp/create project

### LINKED ISSUE #21: User Story: Create and Edit Project
As a user when I visit "/start"
I should see "Choose a category:"
I should see a "Select a category" dropdown box
And I should be able to click on a category
Then I should see a field "Give your project a title:"
And then I should be able to fill in a project title
Then I should see a field "Your permanent residence:"
And then I select my country from a dropdown box.
When I click the button "Save and continue"
I should be redirected to '/user/:id/project/edit'
As an authenticated user
When I visit the projects page
And I click on Edit Project
And I update that project's info
And I click on Update
I should be redirected to that project's show page
And my changes should be reflected

### PR DESCRIPTION
**Pull request by Dpalazzari**
In this PR: 

- Adds project creation functionality.
- Tests project creation functionality.

With Daniel's branch, Closes #21 

This is a work in progress, but you guys should check it out and tell me if you see something I should do differently. 

@Robbie-Smith @kheppenstall @jdconrad89 @drod1000

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@
     # ...rather than:
     #     # => "be bigger than 2"
     expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+    config.backtrace_exclusion_patterns << /\.rvm\/gems/
```
Comment (Dpalazzari) [inline spec/spec_helper.rb]: FYI, this is a line of code Jason told us about that reduces the size of your error messages to just the things you want to see.

**Comment by drod1000** [comment]
Had a chance to review this in person, Iooks good to me!

File: spec/features/projects/user_creates_new_project_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      expect(page).to have_content("20000")
+
+      # As a user when I visit "/start"
+      # I should see "Choose a category:"
+      # I should see a "Select a category" dropdown box
+      # And I should be able to click on a category
+      # Then I should see a field "Give your project a title:"
+      # And then I should be able to fill in a project title
+      # When I click the button "Save and continue"
+      # I should be redirected to '/user/:id/project/edit'
```
Comment (kheppenstall) [inline spec/features/projects/user_creates_new_project_spec.rb]: Let's remove these comments before we merge. @Dpalazzari

File: spec/features/projects/user_creates_new_project_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      category = Category.first
+
+      user = create(:user)
+
+      page.set_rack_session(user_id: user.id)
+
+      visit start_path
+      
+      expect(page).to have_content("Choose a category:")
```
Comment (kheppenstall) [inline spec/features/projects/user_creates_new_project_spec.rb]: Why do we care about having these selectors?

Comment (kheppenstall) [inline spec/features/projects/user_creates_new_project_spec.rb]: @Dpalazzari

File: spec/features/projects/user_creates_new_project_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+      # As a user when I visit "/start"
+      # I should see "Choose a category:"
+      # I should see a "Select a category" dropdown box
+      # And I should be able to click on a category
+      # Then I should see a field "Give your project a title:"
+      # And then I should be able to fill in a project title
+      # When I click the button "Save and continue"
+      # I should be redirected to '/user/:id/project/edit'
+    end
```
Comment (Robbie-Smith) [inline spec/features/projects/user_creates_new_project_spec.rb]: The commented out code needs to be removed.

File: spec/features/projects/user_creates_new_project_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      fill_in 'project[blurb]',    with: 'super duper babooshka kadooshka sploosh'
+      fill_in 'project[goal]',     with: '20000'
+      fill_in 'project[photo]',    with: 'http://dingo.care2.com/pictures/greenliving/1296/1295735.large.jpg'
+      fill_in 'project[deadline]', with: "02/12/2017"
+      click_on "Save and continue"
+
+      expect(current_path).to eq(project_path(Project.last.slug))
+      expect(page).to have_content("babooshka lemon squeezy")
+      expect(page).to have_content("super duper babooshka kadooshka sploosh")
+      expect(page).to have_content("20000")
```
Comment (Robbie-Smith) [inline spec/features/projects/user_creates_new_project_spec.rb]: I think this test should be refactored to provide more clarity as to what is actually being tested.

File: spec/features/projects/user_creates_new_project_spec.rb (hunk 3 above)
Comment (Dpalazzari) [inline spec/features/projects/user_creates_new_project_spec.rb]: Removed!

**Pull request merged**